src/UnitTests/LanguageServerClient/UnitTests.LanguageServerClient/PythonLanguageServerDotNetCore.cs

```diff
@@ -0,0 +1,244 @@
+                    UseShellExecute = false,
+                    CreateNoWindow = true
+                };
+            } else if (File.Exists(serverDllFilePath)) {
+                return new ProcessStartInfo {
+                    FileName = DotNetExeFilePath,
+                    WorkingDirectory = folderPath,
+                    Arguments = '"' + serverDllFilePath + '"',
+                    RedirectStandardInput = true,
+                    RedirectStandardOutput = true,
+                    UseShellExecute = false,
+                    CreateNoWindow = true
+                };
+            } else {
+                Debug.Fail("Could not find language server exe or dll");
+                throw new FileNotFoundException("Could not find language server exe or dll", serverDllFilePath);
+            }
+        }
+
+        private string GetServerLocation() {
+            // For testing, allow to specify via env var the location
+            // of an extracted language server
+            var lsExtractedFolderPath = Environment.GetEnvironmentVariable("PTVS_DOTNETCORE_SERVER_LOCATION");
+            if (Directory.Exists(lsExtractedFolderPath)) {
+                return lsExtractedFolderPath;
+            }
+
+            if (!_joinableTaskContext.IsOnMainThread) {
+                throw new InvalidOperationException("Must be called from main thread.");
+            }
+
+            //var shell = _site.GetService<SVsShell, IVsShell>();
+            //shell.GetProperty((int)__VSSPROPID4.VSSPROPID_LocalAppDataDir, out object localAppDataFolderPath);
+
+            //lsExtractedFolderPath = Path.Combine((string)localAppDataFolderPath, "PythonLanguageServer");
+            //if (!Directory.Exists(lsExtractedFolderPath)) {
+            //    ExtractToFolder(lsExtractedFolderPath);
+            //}
+
+            return lsExtractedFolderPath;
+        }
+
+        public static void ExtractToFolder(string lsExtractedFolderPath) {
+            var lsZipFolderPath = Path.GetDirectoryName(typeof(PythonLanguageServerDotNetCore).Assembly.Location);
+            var lsZipFileName = string.Format(CultureInfo.InvariantCulture, "Python-Language-Server-win-{0}.zip", Environment.Is64BitOperatingSystem ? "x64" : "x86");
+            var lsZipFilePath = Path.Combine(lsZipFolderPath, lsZipFileName);
+
+            if (File.Exists(lsZipFilePath)) {
+                ZipFile.ExtractToDirectory(lsZipFilePath, lsExtractedFolderPath);
+            } else {
+                throw new FileNotFoundException("Python Language Server archive file not found.", lsZipFilePath);
+            }
+        }
+
+        public override void Dispose() {
+            if (_process != null) {
+                try {
+                    _process.Kill();
+                } catch (InvalidOperationException) { } catch (NotSupportedException) { }
+
+                _process.WaitForExit();
+                _process.Dispose();
+                _process = null;
+            }
+        }
```

Comment:
added the Dispose() function above to kill the LS process and catch any exceptions.